diff --git a/protocol/src/main/scala/org/apache/toree/kernel/protocol/v5/Header.scala b/protocol/src/main/scala/org/apache/toree/kernel/protocol/v5/Header.scala
index 1bb3926f..2ad3a83e 100644
--- a/protocol/src/main/scala/org/apache/toree/kernel/protocol/v5/Header.scala
+++ b/protocol/src/main/scala/org/apache/toree/kernel/protocol/v5/Header.scala
@@ -21,13 +21,16 @@ import play.api.libs.json._
 
 case class Header(
   msg_id: UUID,
-  username: String,
-  session: UUID,  // Custom UUID type also used here
+  username: String = "cml",
+  session: UUID = "cml",  // Custom UUID type also used here
   msg_type: String,
   version: String   // Assuming a string since the doc says '5.0'
 )
 
 object Header {
+  // sense does not send `username` and `session`, make sure toree handles these fields missing
+  // this is fixed in sense, but users might be using an older version of cml without this fix
+  implicit def jsonFormat = Json.using[Json.WithDefaultValues].format[Header]
   implicit val headerReads = Json.reads[Header]
   //implicit val headerWrites = Json.writes[Header]
   implicit val headerWriters = new Writes[Header] {
diff --git a/scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala b/scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala
index eb365027..9780ab4d 100644
--- a/scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala
+++ b/scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala
@@ -283,7 +283,9 @@ class ScalaInterpreter(private val config:Config = ConfigFactory.load) extends I
 
      logger.trace(s"Interpreting line: $code")
 
-     val futureResult = interpretAddTask(code, silent)
+     // scala interpreter returns Incomplete on a comment-only code block, which toree would translate
+     // to syntax error. we add an empty code block at the beginning of each code executed to circumvent this.
+     val futureResult = interpretAddTask("{}\n"+code, silent)
 
      // Map the old result types to our new types
      val mappedFutureResult = interpretMapToCustomResult(futureResult)
